packages/@aws-cdk/pipelines/README.md

```diff
@@ -861,6 +861,43 @@ class MyJenkinsStep extends pipelines.Step implements pipelines.ICodePipelineAct
 }
 ```
 
+Another example, adding a lambda step referencing outputs from a stack:
+
+```ts
+class MyLambdaStep extends pipelines.Step implements pipelines.ICodePipelineActionFactory {
+  private stackOutputReference: pipelines.StackOutputReference
+
+  constructor(
+    private readonly function: lambda.Function,
+    stackOutput: CfnOutput,
+  ) {
+    super('MyLambdaStep');
+    this.stackOutputReference = pipelines.StackOutputReference.fromCfnOutput(stackOutput);
+  }
+
+  public produceAction(stage: codepipeline.IStage, options: pipelines.ProduceActionOptions): pipelines.CodePipelineActionFactoryResult {
+
+    stage.addAction(new cpactions.LambdaInvokeAction({
+      actionName: options.actionName,
+      runOrder: options.runOrder,
+      // Map the reference to the variable name the CDK has generated for you.
+      userParameters: {stackOutput: options.stackOutputsMap.mapOutputReference(this.stackOutputReference)},
```

Comment:
This shouldn't compile, since `stackOutputsMap` is optional (should require a `?.` or `!.`), but I'm pretty sure our code translator would balk at that.

Since `ProduceActionOptions` is much more of an output struct than an input struct, you have my permission to make the field required and silence the backwards compatibility checker.

Reply:
I've added the compat error to https://github.com/aws/aws-cdk/pull/23110/commits/5bd7d2cdbb936bd975a25b1afd428ebca100aa1f, is that sufficient?

Reply:
Yes, that is it exactly